dartagnan/src/main/java/com/dat3m/dartagnan/solver/caat4wmm/basePredicates/DynamicDefaultWMMGraph.java

```diff
@@ -0,0 +1,62 @@
+package com.dat3m.dartagnan.solver.caat4wmm.basePredicates;
+
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.Edge;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.RelationGraph;
+import com.dat3m.dartagnan.verification.model.EventData;
+import com.dat3m.dartagnan.wmm.relation.Relation;
+import com.dat3m.dartagnan.wmm.utils.Tuple;
+import org.sosy_lab.java_smt.api.Model;
+import org.sosy_lab.java_smt.api.SolverContext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+// A default implementation for any encoded relation, e.g. base relations or non-base but cut relations.
+public class DynamicDefaultWMMGraph extends MaterializedWMMGraph {
+    private final Relation relation;
+
+    public DynamicDefaultWMMGraph(Relation rel) {
+        this.relation = rel;
+    }
+
+    @Override
+    public List<RelationGraph> getDependencies() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void repopulate() {
```

Comment:
The second call can be faster because `model.getEventList()` just contains executed events while if we would always take the first branch, we would potentially iterate over all the domain (which might contain pairwise exclusive events), right?

Reply:
We won't iterate over pairwise exclusive events, because those are usually removed from `maxTupleSet`.
The first branch is "edge"-centric: It iterates over all possible edges(tuples) and checks if the corresponding events are executed.
The second branch is "event"-centric: It iterates over all possible executed event-pairs and sees if there is an edge between them.
In some cases the first approach is better and in some the second.